File: AulaTPII/F217/GoF/Estruturais/Adapter/Adapter01.js
// Interface do Cliente
class Itarget{
    request(){}
}

// Cliente
class Cliente{
    constructor(target){
        this.target = target;
    }

    makeRequest(){
        console.log("Fazendo Uma Requisição");
        this.target.request();
    }

}


// Serviço Existente (Adaptee)
class Adaptee{
    SpecificRequest(){
        console.log("Requisição Especifica do Adaptee");
    }
}

// Adaptador (Adapter)
class Adapter{
    constructor(adaptee){
        this.adaptee = adaptee;
    }

    request(){
        this.adaptee.SpecificRequest();
    }
}

// Utilizando o adaptador

const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const cliente = new Cliente(adapter);

cliente.makeRequest();

File: AulaTPII/F217/GoF/Estruturais/Adapter/Adapter01.py
# Interface do Cliente
class ITarget:
    def request(self):
        pass

# Cliente
class Cliente:
    def __init__(self, target):
        self.target = target

    def makeRequest(self):
        print("Fazendo uma requisição")
        self.target.request()

# Serviço Existente (Adaptee)
class Adaptee:
    def SpecificRequest(self):
        print("Requisição Especifica do Adaptee")

# Adaptador (Adapter)
class Adapter(ITarget):
    def __init__(self, adaptee):
        self.adaptee = adaptee

    def request(self):
        self.adaptee.SpecificRequest()

# Utilizando o Adaptador ##########
adaptee = Adaptee()
adapter = Adapter(adaptee)
client = Cliente(adapter)

client.makeRequest()
